docs/NEW_NEAR_CLI_INTERFACE.md

```diff
@@ -0,0 +1,114 @@
+# New NEAR CLI interface
+
+`NEAR CLI` is using `extensions` to saticfy all groups of users.
+- `Core NEAR CLI` commands should be usefull for all groups of users.
+- `Core NEAR CLI` is a single binary, there is no extensions that are installed `by default`.
+- Each extension is a separate binary that can be executed from `NEAR CLI`.
+- Extensions are used by a particular group or several groups of users.
+- Extensions are not composable (you should not create extensions for extensions).
+
+`NEAR CLI` is built for:
+- NEAR `dApp developers`, who build smart-contracts, UIs, and tooling on NEAR.
+- `Tech-savvy` people automating their routines.
+- `Validators`
+
+`NEAR CLI UX principles`
+- All altering actions should have a confirmation step with an option to skip confirmation with an explicit command line parameter (e.g. `send` at the end of the command)
+- All direct children commands of a single parent command should be aligned (either represent an action or a resource, but never a mix of those on the same hierarchy level): `contract` -> `state` (resource) and `contract` -> `deploy` (action) are not aligned, so it should be either `contract` -> `get-state` + `contract` -> `deploy` or `contract` -> `state` -> `view` + `contract` -> `code` -> `deploy`
+- Interactive mode should look like: `command - description`. It will help people to learn the commands.
+
+
+## `Core NEAR CLI` commands
+```txt
+import-account
+    - from-near-wallet
+    - from-seed-phrase <seed-phrase>
+    - from-ledger
+    - from-private-key <private-key>
+
+create-account
+    - implicit
+    - subaccount <master-account> <new-subaccount-id>
+
+manage-account <accoundId>
```

Comment:
This whole thing is tough, and there is always going to be a tradeoff, so treat my comments as inputs, not the action items.

I feel we diverged from your original proposal of building a resource-oriented design. Should we introduce a top-level `account` command?

* account
  * import
  * create
  * manage

This way it aligns better with `explorer`, `transaction-construction`, etc extensions which are resources, not actions.

There is a problem, however; we can argue that `explorer` can also have `account` resource, and given that almost everything in NEAR is somehow related to accounts, it becomes hard to draw a boundary between those. E.g. is contract manipulation commands are part of account management? (yes and no...)

Reply:
Yes, I was thinking about this design.
And yes, there are tradeoffs. Let's not forget, that our CLI will be used by humans. They will come and ask "Where is my `near send` command".

Here is one more option for the layout of the account commands.
```txt
account
    - import
        - from-near-wallet
        - from-seed-phrase <seed-phrase>
        - from-ledger
        - from-private-key <private-key>
    - create
        - implicit
        - subaccount <master-account> <new-subaccount-id>
    - manage <accoundId>
        - state
            - view
            - transfer <sender> <reciever> <amount>
        - keys
            - view
            - add
            - delete <public-key>
        - contract
            - deploy-code <wasm-path>
            - get-code-checksum
            - get-state
            - call-view-method
            - call-change-method
    - delete
```
In this case, we are avoiding unnecessary `manage` and `get` words in command names. But we are adding one more nested layer.
And I'm still not happy with `near`-> `account` -> `manage` -> `state` -> `view` and `transfer`

Reply:
I agree that we need to separate `account` into several entities. `contract` is a first candidate for me, since it's not a part of the account, it just belongs to the account.
We can have something similar to `my-account`, probably we can put `actions` there, since they require a key to be executed. It can look like this:
```txt
actions
    - account
        - create
        - delete
    - key
        - add
        - delete
    - transfer
    - stake
    - deploy-contract
    - function-call
```

But with this design, we will put `function-call`s, `keys` and `transfers` under one roof :/

Reply:
> But with this design, we will put function-calls, keys and transfers under one roof :/

This is not fun. This is too implementation-driven, and I feel that `actions` is a catch-all name. I wish I could spend more time thinking through it... So far I am not happy with the explored options :disappointed: